controllers/perconaservermysql_controller.go

```diff
@@ -143,7 +144,27 @@ func (r *PerconaServerMySQLReconciler) doReconcile(
 	if err := r.cleanupOutdated(ctx, cr); err != nil {
 		return errors.Wrap(err, "cleanup outdated")
 	}
+	if err := r.ensureOperatorVersion(ctx, cr); err != nil {
+		return errors.Wrap(err, "ensure operator version")
+	}
+
+	return nil
+}
+
+func (r *PerconaServerMySQLReconciler) ensureOperatorVersion(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL) error {
+	if cr.Spec.UpgradeOptions.Apply == "" ||
+		cr.Spec.UpgradeOptions.Apply == apiv1alpha1.UpgradeStrategyDisabled ||
+		cr.Spec.UpgradeOptions.Apply == apiv1alpha1.UpgradeStrategyNever {
+		return nil
+	}
+
+	log := log.FromContext(ctx).WithName("ensureOperatorVersion")
 
+	version, err := versionservice.GetExactVersion(ctx, cr, r.ServerVersion)
+	if err != nil {
+		return errors.Wrap(err, "get exact version")
+	}
+	log.Info("got version", "versions", version.String())
```

Comment:
but what are we doing with versions? if it's like this because of lack of smart update, I think we can just patch images and rollout restart. what do you think @hors ?

Reply:
under this PR lets just get images from VS and use it in CR (rewrite existing one)

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/151/commits/310d34f7f7ac320526be0346fc113e938bcaf06c